rest/util.go

```diff
@@ -126,11 +126,13 @@ func compareEtag(etag, baseEtag string) bool {
 	if etag == "" {
 		return false
 	}
-	if etag == baseEtag {
-		return true
-	}
-	if l := len(etag); l == len(baseEtag)+2 && l > 3 && etag[0] == '"' && etag[l-1] == '"' && etag[1:l-1] == baseEtag {
-		return true
+	if strings.HasPrefix(etag, "W/") {
+		if etag[2:] == baseEtag {
+			return true
+		}
+		if l := len(etag); l == len(baseEtag)+4 && l > 4 && etag[2] == '"' && etag[l-1] == '"' && etag[3:l-1] == baseEtag {
+			return true
+		}
```

Comment:
> I think that quotes are optional for rest-layer. Thanks next condition will match the qoutes as well.

Right. Maybe we should simplify the code here a bit to make it more readable.

```go
if etag == "" {
	return false
}
cmpEtag := strings.StripPrefix(etag, "W/")
cmpEtag = strings.StripPrefix(etag, `"`)
cmpEtag = strings.StripSuffix(etag, `"`)
return cmpEtag == baseEtag
``

Reply:
PS! You don't need to do this if you don't feel like it. Maybe I can do it ;-) I kind of want to benchmark it as well, just for fun.

Reply:
Yes, I guess we can simplify it a little more. I was just following already written code style.

I don't expect any significant overhead of computing the ETags, compared to actual network latency. You can write some benchmarks of you want though.

Reply:
> . I was just following already written code style.

It's OK, don't worry abuout it. It was nice work!

The benchmarks would only for my own case, and just for fun. It's easy to assume one thing is faster than another without being right, and this things migh affect the code one write in general.

I think we should do the cleanups either way, but I can take care of it.